submodules/eks/k8s.tf

```diff
@@ -1,25 +1,47 @@
+locals {
+  run_setup = var.bastion_info != null || var.eks.public_access.enabled ? 1 : 0
+}
+
 module "k8s_setup" {
-  count = var.bastion_info != null || var.eks.public_access.enabled ? 1 : 0
+  count = local.run_setup
 
-  source        = "../k8s"
-  ssh_key       = var.ssh_key
-  bastion_info  = var.bastion_info
-  network_info  = var.network_info
-  eks_info      = local.eks_info
-  cluster_setup = true
+  source       = "../k8s"
+  ssh_key      = var.ssh_key
+  bastion_info = var.bastion_info
+  network_info = var.network_info
+  eks_info     = local.eks_info
 
   depends_on = [aws_eks_addon.vpc_cni, null_resource.kubeconfig]
 }
 
-module "calico_setup" {
-  count = var.bastion_info != null || var.eks.public_access.enabled ? 1 : 0
+resource "null_resource" "run_k8s_pre_setup" {
```

Comment:
Can we run this one as part of the `k8s_setup` otherwise `k8s_setup` is only rendering the templates

Reply:
I had it like that after the initial change, but it seemed like kind of a strange division of labor to have the script running there, and then arbitrarily (if still for a reason) running it again outside of the module, including all the bits to run it correctly externally.